packages/buendia-utils/data/usr/bin/buendia-init

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+LOG=/var/log/buendia/buendia-init.log
+HWCLOCK_FILE=/var/log/buendia/hwclock.final
+
+# Ensure no dependency on any environment variables, even PATH.
+export PATH=/sbin:/usr/sbin:/bin:/usr/bin
+
+action="$1"
+
+# Reports the date and time on the system clock to nanosecond precision.
+function nanodate() {
+    date +'%Y-%m-%d %H:%M:%S.%N'
+}
+
+# Saves the time reported by the hardware clock (for use during shutdown).
+function save_hwclock() {
+    hwclock -u > $HWCLOCK_FILE
+}
+
+# Checks the hardware clock to see if it has moved backwards since the last
+# shutdown, and if so, disables the Buendia API to prevent database corruption.
+#
+# Loss of battery power to the internal hardware clock can cause the clock to
+# reset to a time in the past.  If data is written to the database that is
+# timestamped in the past, it is not only invalid, but can also make Buendia
+# clients unusable.  Hence, in this eventuality, we disable the Buendia API.
+function check_hwclock() {
+    now=$(echo $(/sbin/hwclock -u))
+    final=$(echo $(cat $HWCLOCK_FILE))
+
+    if [[ "$now" < "$final" ]]; then
+        systemctl mask tomcat7
+        buendia-led pink 100  # system has entered an error state
```

Comment:
@zestyping does `buendia-led` do anything anymore?

Reply:
Not at the moment.  These `buendia-led` calls crept in because I have a local experiment that controls the NUC LED and it's pretty sweet, but isn't ready for production yet.

`buendia-led` is written to exit with success no matter what, so it's harmless here.